packages/framework/src/Component/Router/FriendlyUrl/FriendlyUrlFactory.php

```diff
@@ -19,4 +21,29 @@ public function create(
     ): FriendlyUrl {
         return new FriendlyUrl($routeName, $entityId, $domainId, $slug);
     }
+
+    /**
+     * @param string $routeName
+     * @param int $entityId
+     * @param string $entityName
+     * @param int $domainId
+     * @param int|null $indexPostfix
+     * @return \Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrl|null
+     */
+    public function createIfValid(
+        string $routeName,
+        int $entityId,
+        string $entityName,
+        int $domainId,
+        int $indexPostfix = null
+    ):?FriendlyUrl {
```

Comment:
there should be a space between ":" and "?", right? We should have a fixer for that :slightly_smiling_face: